### PR TITLE
Hardware MVP

### PR DESCRIPTION
**Pull request by zsarnett**
## Breaking change

Moves Reboot and Shutdown to overflow (not really breaking)

## Proposed change

CleanShot 2022-05-24 at 23 55 30

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/hardware/ha-config-hardware.ts [hunk 1]
```diff
@@ -50,6 +56,27 @@ class HaConfigHardware extends LitElement {
...
+    if (
+      isComponentLoaded(this.hass, "hardware") &&
+      this._hardwareInfo?.hardware.length
+    ) {
+      const boardData = this._hardwareInfo!.hardware[0];
+
+      board = boardData.board.model;
+      hostName = boardData.name;
+      documentationURL = boardData.url;
+      imageURL =
```
Comment (zsarnett) [inline src/panels/config/hardware/ha-config-hardware.ts:73]: Still need to build this from the board information

File: src/panels/config/hardware/ha-config-hardware.ts [hunk 2]
```diff
@@ -76,54 +117,46 @@ class HaConfigHardware extends LitElement {
...
+                            <ha-clickable-list-item
+                              .href=${documentationURL}
+                              openNewTab
+                              twoline
+                              hasMeta
+                            >
+                              <span>Documentation</span>
+                              <span slot="secondary"
+                                >Find extra information on your device</span
+                              >
```
Comment (zsarnett) [inline src/panels/config/hardware/ha-config-hardware.ts:156]: Needs to be Localized

**Comment by spacegaier** [comment]
> Moves Reboot and Shutdown to overflow (not really breaking)

Just curios: Is there a reason for that? Hiding things in sub menus without a real need always introduces extra clicks. On a hardware dedicated page, those buttons seem rather fitting.

File: src/panels/config/hardware/ha-config-hardware.ts [hunk 3]
```diff
@@ -50,6 +56,27 @@ class HaConfigHardware extends LitElement {
   }
 
   protected render(): TemplateResult {
+    let board: string | undefined;
+    let hostName: string | undefined;
+    let imageURL: string | undefined;
+    let documentationURL: string | undefined;
+
+    if (
+      isComponentLoaded(this.hass, "hardware") &&
```
Comment (bramkragten) [inline src/panels/config/hardware/ha-config-hardware.ts:65]: No need for this check, `this._hardwareInfo` will be undefined if there is no hardware integration

**Comment by zsarnett** [comment]
> Just curios: Is there a reason for that?

The plans for this page will continue to grow and going ahead and placing them here will be the best instead of waiting. Something we should have done before

File: src/panels/config/hardware/ha-config-hardware.ts [hunk 4]
```diff
@@ -160,10 +202,13 @@ class HaConfigHardware extends LitElement {
     });
 
     if (!confirmed) {
-      button.progress = false;
       return;
     }
 
+    showToast(this, {
+      message: this.hass.localize("ui.panel.config.hardware.rebooting_host"),
```
Comment (bramkragten) [inline src/panels/config/hardware/ha-config-hardware.ts:209]:
```suggestion
      message: this.hass.localize("ui.panel.config.hardware.rebooting_host"),
      duration: 0,
```

File: src/panels/config/hardware/ha-config-hardware.ts [hunk 5]
```diff
@@ -42,14 +48,35 @@ class HaConfigHardware extends LitElement {
...
+    let documentationURL: string | undefined;
+
+    if (this._hardwareInfo?.hardware.length) {
+      const boardData = this._hardwareInfo!.hardware[0];
+
+      boardId = boardData.board.hassio_board_id;
+      boardName = boardData.name;
+      documentationURL = boardData.url;
+      imageURL = hardwareBrandsUrl({
+        domain: "boards",
```
[the remark this answers is not in the document]
Comment (bramkragten) [reply src/panels/config/hardware/ha-config-hardware.ts:71]: Maybe make `category`/`type` or something?

File: src/panels/config/hardware/ha-config-hardware.ts [hunk 6]
```diff
@@ -42,14 +48,35 @@ class HaConfigHardware extends LitElement {
...
+        name: `${boardData.board.manufacturer}_${boardData.board.model}`,
```
[the remark this answers is not in the document]
Comment (zsarnett) [reply src/panels/config/hardware/ha-config-hardware.ts:72]: Im not sure that we can guarentee that manufacturer and model will be the defining points for this. Zwave devices for example. they may have other features to define the name

But I think we can do this for now and pivot

**Review by bramkragten — APPROVED** [review]

**Pull request merged**